tests/deprecated_api/test_remove_1-8.py

```diff
@@ -500,3 +500,35 @@ def on_before_accelerator_backend_setup(self, *args, **kwargs):
         " and will be removed in v1.8"
     ):
         trainer.fit(model)
+
+
+def test_v1_8_0_datamodule_checkpointhooks(tmpdir):
```

Comment:
I was thinking this allows more proactive warnings - we don't have to wait for if and when checkpoint save/load happens, but rather warn whenever a datamodule with these hooks is being used

Reply:
I'm proposing this to reduce the test runtime and scope.

Relying on a `fit` call to test is a bit lazy and a pattern we've abused in the past and are trying to change now.

Reply:
oh I mean I purposely put the deprecation warnings inside `_run` so that the user can get more proactive warnings - gives up the specificity of conditioning and timing the warnings on if and when save/load actually happens, but this way warns whenever a datamodule with these hooks is being used.

lmk if less proactively and more specifically warning users only if and when save/load occurs is preferred over this though

Reply:
My request is that we write the test like this:

```python
def test_warning():
    trainer = Mock()
    
    trainer.datamodule = CustomBoringDataModuleSave()
    with pytest.deprecated_call(match=...)
        _check_datamodule_checkpoint_hooks(trainer)

    same for CustomBoringDataModuleLoad
    ...
```

Which would run much faster than 2 `trainer.fit` calls

Reply:
ah got it